src/passes/J2CLItableMerging.cpp

```diff
@@ -104,20 +104,33 @@ struct J2CLItableMerging : public Pass {
         continue;
       }
 
-      auto type = heapType.getStruct();
-      if (typeNameInfo.fieldNames.empty() ||
-          !typeNameInfo.fieldNames[0].equals("vtable")) {
-        continue;
-      }
-      if (typeNameInfo.fieldNames.size() < 1 ||
-          !typeNameInfo.fieldNames[1].equals("itable")) {
+      if (typeNameInfo.fieldNames.empty()) {
         continue;
       }
 
-      auto vtabletype = type.fields[0].type.getHeapType();
-      auto itabletype = type.fields[1].type.getHeapType();
+      // The vtable may either be the first field or the custom descriptor.
+      std::optional<HeapType> vtabletype;
+      std::optional<HeapType> itabletype;
```

Comment:
If these would be empty, then we'll just `continue` to the next loop iteration anyway, so I don't see a reason to make them optional.
```suggestion
      HeapType vtabletype;
      HeapType itabletype;
```